services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -960,19 +956,31 @@ public static Response redirectAfterSuccessfulFlow(KeycloakSession session, Real
 
     }
 
-    public static String getSessionIdFromSessionCookie(KeycloakSession session) {
+    /**
+     * @param session keycloak session
+     * @param sessionId in plain-text
+     * @return true if sessionId matches with the session from KEYCLOAK_SESSION_COOKIE
+     */
+    public static boolean compareSessionIdWithSessionCookie(KeycloakSession session, String sessionId) {
+        if (sessionId == null) {
+            throw new IllegalArgumentException("Not expected to provide null sessionId");
+        }
```

Comment:
Not sure if possible, but why not use the `assert` keyword here? See: https://www.baeldung.com/java-assert

Reply:
There is no reason to not use `assert` . It is probably slightly better since it is 1 line instead of 3, so I've just updated to use that.